### ISSUE TITLE
Nestjs support

### ISSUE THREAD

**Issue by kodeine** [mercurius-js/mercurius#386]
Hello, is this supported with nestjs? Any example i can take a look at ?

**Comment by mcollina** [comment]
I have never tried to run this with Nestjs. It _might_ work

**Comment by smolinari** [comment]
I'd just like to add, with the neglect Apollo Server has had the past couple of years and seeing this project looking pretty lively and interesting with decent docs, AND being a NestJS user, this issue caught my eye directly. Apollo Server's Fastify support is broken AFAIK or it requires a good bit of patching to get working and overall the situation with Apollo Server is distressing, and almost disastrous. I have no confidence in that package or the Apollo team in producing something new and of real value for Apollo Server any more. 

If someone would create an integration of Mercurius with NestJS, I'd switch immediately. 

Scott

**Comment by mcollina** [comment]
If anybody would like to work on an adapter inside this github organization, let us know and we'll set up a repo!

**Comment by Davide-Gheri** [comment]
Hey all, I just started working on it:
https://github.com/Davide-Gheri/nestjs-mercurius

It is still an early work in progress but a simple GraphQL server "works" (queries/mutations/field resolvers) with code first approach.

I reused a lot of the official **@nestjs/graphql** module, basically writing only a new MercuriusModule that register mercurius to the Nestjs Fastify instance.

I managed to make **Loaders** work defining some custom decorators but I'm not completely satisfied with the result.

I still have to test Subscriptions and think about how to implement Federation, any help and feedback is much appreciated!
